packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -140,3 +167,68 @@ export function unstable_httpSubscriptionLink<
     };
   };
 }
+
+/**
+ * We wrap EventSource so that is can be reinitialized with new options
+ */
+class EventSourceWrapper implements Partial<EventSource> {
+  private es: EventSource;
+
+  private listeners: Partial<Record<keyof EventSourceEventMap, any[][]>> = {};
+
+  constructor(url: string, options: EventSourceInit | undefined) {
+    this.es = new EventSource(url, options);
+  }
+
+  restart(url: string, options: EventSourceInit | undefined) {
+    this.es.close();
```

Comment:
you probably wanna prevent this from emitting any events

Reply:
Agreed, haven't seen any negative effects, but no sense in risking it

Reply:
This one is done, removed all known listeners before closing the connection